src/Adapter/Product/CommandHandler/UpdateProductPricesHandler.php

```diff
@@ -0,0 +1,221 @@
+            $this->validateField($product, 'price', ProductConstraintException::INVALID_PRICE);
+        }
+
+        if (null !== $unitPrice) {
+            $this->setUnitPriceInfo($product, $unitPrice, $price);
+        }
+
+        if (null !== $command->getUnity()) {
+            $product->unity = $command->getUnity();
+        }
+
+        if (null !== $command->getEcotax()) {
+            $product->ecotax = (float) (string) $command->getEcotax();
+            $this->validateField($product, 'ecotax', ProductConstraintException::INVALID_ECOTAX);
+        }
+
+        $taxRulesGroupId = $command->getTaxRulesGroupId();
+
+        if (null !== $taxRulesGroupId) {
+            $product->id_tax_rules_group = $taxRulesGroupId;
+            $this->validateField($product, 'id_tax_rules_group', ProductConstraintException::INVALID_TAX_RULES_GROUP_ID);
+            $this->assertTaxRulesGroupExists($taxRulesGroupId);
+        }
+
+        if (null !== $command->isOnSale()) {
+            $product->on_sale = $command->isOnSale();
+        }
+
+        if (null !== $command->getWholesalePrice()) {
+            $product->wholesale_price = (float) (string) $command->getWholesalePrice();
+            $this->validateField($product, 'wholesale_price', ProductConstraintException::INVALID_WHOLESALE_PRICE);
+        }
+    }
+
+    /**
+     * @param int $taxRulesGroupId
+     *
+     * @throws ProductConstraintException
+     * @throws ProductException
+     */
+    private function assertTaxRulesGroupExists(int $taxRulesGroupId): void
+    {
+        // 0 is valid value meaning no tax rules group is applied.
+        if (0 === $taxRulesGroupId) {
+            return;
+        }
+
+        $code = ProductConstraintException::INVALID_TAX_RULES_GROUP_ID;
+
+        try {
+            $taxRulesGroup = new TaxRulesGroup($taxRulesGroupId);
+            if (!$taxRulesGroup->id) {
+                throw new ProductConstraintException(
+                    sprintf(
+                        'Invalid tax rules group id "%s". Group doesn\'t exist',
+                        $taxRulesGroupId
+                    ),
+                    $code
+                );
+            }
+        } catch (PrestaShopException $e) {
+            throw new ProductException(
+                sprintf(
+                    'Error occurred when trying to load tax rules group #%s for product',
+                    $taxRulesGroupId
+                ),
+                0,
+                $e
+            );
+        }
+    }
+
+    /**
+     * @param Product $product
+     * @param Number $unitPrice
+     * @param Number $price
+     *
+     * @throws ProductConstraintException
+     */
+    private function setUnitPriceInfo(Product $product, Number $unitPrice, ?Number $price): void
+    {
+        $this->validateUnitPrice($unitPrice);
+
+        if ($unitPrice->equals(new Number('0'))) {
+            return;
+        }
+
+        if (null === $price) {
```

Comment:
if its a partial update and price is not provided, we fallback to Product price from db. If its also 0 we throw exception, because in this case the unit_price and ratio can only be 0

Reply:
can you add this information into the code as comment ?

Reply:
Not sure. It doesn't look right if we allow resetting price when setting unit_price to 0 (it could be vice versa tho). So I think error is fine, everything else we can do in form validation/js side.